### PR TITLE
Migrate specs update to updatecli

### PR DESCRIPTION
**Pull request by reakaleek**
## What

Migrate https://github.com/elastic/apm/blob/main/.ci/Jenkinsfile to the consumer using updatecli and GH Actions workflows.

## Why

Jenkins 2 GH Actions Migration

## How to test

```shell
export GITHUB_TOKEN="<your-personal-access-token>"
export GIT_USER="FooBar"
export GIT_EMAIL="foo@bar.com"
updatecli --config .ci/update-json-specs.yml diff
updatecli --config .ci/update-gherkin-specs.yml diff
```

### CODE REVIEW

File: .ci/update-gherkin-specs.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+name: update-gherkin-specs
+title: Update Gherkin specs
+
```
Comment (v1v) [inline .ci/update-gherkin-specs.yml:3]: what do you think adding the fielld `pipelineid: 'update-gherkin-specs'` so the branch name is easy to recognise what's for?

Comment (reakaleek) [reply .ci/update-gherkin-specs.yml:3]: sure, will do also in the other PRs

Comment (reakaleek) [reply .ci/update-gherkin-specs.yml]: https://github.com/elastic/apm-agent-php/pull/889/commits/bd1cbcc032aec198c508e3b74403f7b3adda7198

**Review by v1v — APPROVED** [review]

File: .ci/updatecli.d/update-gherkin-specs.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/gherkin-specs/user_agent.feature
+
+actions:
+  pr:
+    kind: "github/pullrequest"
+    scmid: default
+    spec:
+      title: Update Gherkin specs
```
Comment (v1v) [inline .ci/updatecli.d/update-gherkin-specs.yml:56]:
```suggestion
    title: '[Automation] Update Gherkin specs'
    spec:
```

File: .ci/updatecli.d/update-json-specs.yml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  wildcard_matcher_tests.json:
+    kind: file
+    spec:
+      file: https://raw.githubusercontent.com/elastic/apm/main/tests/agents/json-specs/wildcard_matcher_tests.json
+actions:
+  pr:
+    kind: "github/pullrequest"
+    scmid: default
+    spec:
+      title: Update JSON specs
```
Comment (v1v) [inline .ci/updatecli.d/update-json-specs.yml:59]:
```suggestion
    title: '[Automation] Update JSON specs'
    spec:
```

**Review by v1v — CHANGES_REQUESTED** [review]

**Pull request merged**